aries_cloudagent/resolver/base.py

```diff
@@ -75,16 +97,19 @@ def supports(self, did: Union[str, DID]) -> bool:
 
         return False
 
-    async def resolve(self, profile: Profile, did: Union[str, DID]) -> DIDDocument:
+    async def resolve(self, profile: Profile, did: Union[str, DID]) -> Resolution:
         """Resolve a DID using this resolver."""
         py_did = DID(did) if isinstance(did, str) else did
 
         if not self.supports(py_did):
             raise DIDMethodNotSupported(
                 f"{self.__class__.__name__} does not support DID method {py_did.method}"
             )
-
+        previous_time = datetime.utcnow()
```

Comment:
`resolution_start_time` might be clearer than `previous_time`; I was a bit lost what this value was used for before inspecting the `_retrieve_resolver_metadata` call.